service/src/main/java/io/camunda/service/AuthorizationServices.java

```diff
@@ -132,10 +132,31 @@ public CompletableFuture<AuthorizationRecord> deleteAuthorization(final long aut
     return sendBrokerRequest(brokerRequest);
   }
 
+  public CompletableFuture<AuthorizationRecord> updateAuthorization(
+      final UpdateAuthorizationRequest request) {
+    final var brokerRequest =
+        new BrokerAuthorizationRequest(AuthorizationIntent.UPDATE)
+            .setAuthorizationKey(request.authorizationKey())
+            .setOwnerId(request.ownerId())
+            .setOwnerType(request.ownerType())
+            .setResourceId(request.resourceId())
+            .setResourceType(request.resourceType())
+            .setPermissions(request.permissions());
+    return sendBrokerRequest(brokerRequest);
+  }
+
   public record CreateAuthorizationRequest(
       String ownerId,
       AuthorizationOwnerType ownerType,
       String resourceId,
       AuthorizationResourceType resourceType,
       Set<PermissionType> permissions) {}
+
+  public record UpdateAuthorizationRequest(
+      long authorizationKey,
+      String ownerId,
+      AuthorizationOwnerType ownerType,
+      String resourceId,
+      AuthorizationResourceType resourceType,
+      Set<PermissionType> permissions) {}
```

Comment:
🤔 I wonder if it's easier to just put the `CreateAuthorizationRequest` in here instead of copying all the properties. But the naming won't make sense that way so maybe it'll cause more confusion than good.

Reply:
I'll keep it like this, more clear